flink-runtime/src/test/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtilsTest.java

```diff
@@ -0,0 +1,323 @@
+		});
+	}
+
+	@Test
+	public void testConfigJvmMetaspaceSize() throws Exception {
+		final MemorySize jvmMetaspaceSize = MemorySize.parse("50m");
+
+		Configuration conf = new Configuration();
+		conf.setString(TaskManagerOptions.JVM_METASPACE, jvmMetaspaceSize.getMebiBytes() + "m");
+
+		validateInAllConfigurations(conf, taskExecutorResourceSpec -> assertThat(taskExecutorResourceSpec.getJvmMetaspaceSize(), is(jvmMetaspaceSize)));
+	}
+
+	@Test
+	public void testConfigJvmOverheadSizeAbsolute() throws Exception {
+		final MemorySize jvmOverheadSize = MemorySize.parse("50m");
+
+		Configuration conf = new Configuration();
+		conf.setString(TaskManagerOptions.JVM_OVERHEAD_MAX, jvmOverheadSize.getMebiBytes() + "m");
+		conf.setString(TaskManagerOptions.JVM_OVERHEAD_MIN, jvmOverheadSize.getMebiBytes() + "m");
+
+		validateInAllConfigurations(conf, taskExecutorResourceSpec -> assertThat(taskExecutorResourceSpec.getJvmOverheadSize(), is(jvmOverheadSize)));
+	}
+
+	@Test
+	public void testConfigJvmOverheadSizeFraction() throws Exception {
+		final MemorySize minSize = MemorySize.parse("50m");
+		final MemorySize maxSize = MemorySize.parse("200m");
+		final float fraction = 0.2f;
+
+		Configuration conf = new Configuration();
+		conf.setString(TaskManagerOptions.JVM_OVERHEAD_MAX, maxSize.getMebiBytes() + "m");
+		conf.setString(TaskManagerOptions.JVM_OVERHEAD_MIN, minSize.getMebiBytes() + "m");
+		conf.setFloat(TaskManagerOptions.JVM_OVERHEAD_FRACTION, fraction);
+
+		validateInAllConfigurations(conf, taskExecutorResourceSpec -> {
+			assertThat(taskExecutorResourceSpec.getJvmOverheadSize().getBytes(), greaterThanOrEqualTo(minSize.getBytes()));
+			assertThat(taskExecutorResourceSpec.getJvmOverheadSize().getBytes(), lessThanOrEqualTo(maxSize.getBytes()));
+			assertThat(taskExecutorResourceSpec.getJvmOverheadSize(), anyOf(
+				is(minSize),
+				is(maxSize),
+				is(taskExecutorResourceSpec.getTotalProcessMemorySize().multiply(fraction))
+			));
+		});
+	}
+
+	@Test
+	public void testConfigTotalFlinkMemorySize() throws Exception {
+		final MemorySize totalFlinkMemorySize = MemorySize.parse("1g");
+
+		Configuration conf = new Configuration();
+		conf.setString(TaskManagerOptions.TOTAL_FLINK_MEMORY, totalFlinkMemorySize.getMebiBytes() + "m");
+
+		TaskExecutorResourceSpec taskExecutorResourceSpec = TaskExecutorResourceUtils.resourceSpecFromConfig(conf);
+		assertThat(taskExecutorResourceSpec.getTotalFlinkMemorySize(), is(totalFlinkMemorySize));
+	}
+
+	@Test
+	public void testConfigTotalProcessMemorySize() throws Exception {
+		final MemorySize totalProcessMemorySize = MemorySize.parse("1g");
+
+		Configuration conf = new Configuration();
+		conf.setString(TaskManagerOptions.TOTAL_PROCESS_MEMORY, totalProcessMemorySize.getMebiBytes() + "m");
+
+		TaskExecutorResourceSpec taskExecutorResourceSpec = TaskExecutorResourceUtils.resourceSpecFromConfig(conf);
+		assertThat(taskExecutorResourceSpec.getTotalProcessMemorySize(), is(totalProcessMemorySize));
```

Comment:
Same here

Reply:
Same for `TaskExecutorResourceSpec#getProcessFlinkMemorySize`.